src/engine/Service.cpp

```diff
@@ -120,25 +120,39 @@ ProtoResult Service::computeResult([[maybe_unused]] bool requestLaziness) {
             << ", target: " << serviceUrl.target() << ")" << std::endl
             << serviceQuery << std::endl;
 
-  cppcoro::generator<std::span<std::byte>> jsonByteResult = getResultFunction_(
+  HttpOrHttpsResponse response = getResultFunction_(
       serviceUrl, cancellationHandle_, boost::beast::http::verb::post,
       serviceQuery, "application/sparql-query",
       "application/sparql-results+json");
 
   std::basic_string<char, std::char_traits<char>,
                     ad_utility::AllocatorWithLimit<char>>
       jsonStr(_executionContext->getAllocator());
-  for (std::span<std::byte> bytes : jsonByteResult) {
+  for (std::span<std::byte> bytes : response.body_) {
     jsonStr.append(reinterpret_cast<const char*>(bytes.data()), bytes.size());
     checkCancellation();
   }
 
-  // Parse the received result.
   auto throwErrorWithContext = [&jsonStr](std::string_view sv) {
     throw std::runtime_error(absl::StrCat(
         sv,
         " First 100 bytes: ", std::string_view{jsonStr.data()}.substr(0, 100)));
   };
+
+  // Verify status and content-type of the response.
+  if (response.status_ != boost::beast::http::status::ok) {
+    throwErrorWithContext(absl::StrCat(
+        "Service respondet with: ", static_cast<int>(response.status_), ", ",
+        std::string(boost::beast::http::obsolete_reason(response.status_)),
+        "."));
+  }
+  if (response.contentType_ != "application/json") {
+    throwErrorWithContext(absl::StrCat(
+        "Expected content-type 'application/json', but received: '",
+        response.contentType_, "'."));
```

Comment:
I was confused at first too. As it turns out, QLever **used** `createJsonResponse()` in `HttpUtils` for all outgoing json msgs (our result-types too), where said function sets the content-type to standard json.
As both json-types use `sendStreamableResponse` since yesterdays merge, this won't be a problem anymore.

Reply:
Oh wow, now I understand, you mea, QLever always sent the wrong mediaType?

... 
That's why wewrite unit tests (`Server.cpp` has almost none, therefore its buggy as you can see.